### PR TITLE
[cmake] Create one command for many resources in ADD_QGIS_RESOURCES

### PR DESCRIPTION
**Pull request by nyalldawson**
Instead of creating a single command for EVERY resource, just build up a single command for ALL the resources.

This avoids thousands of lines of output when compiling, where we get one line per resource file. Eg:

    [890/11818 452.2/sec] Generating ../../output/data/svg/sport/sport_snooker.svg
    [891/11818 451.8/sec] Generating ../../output/data/svg/sport/sport_soccer.svg

Since we're installing thousands and thousands of resources (eg the svg symbols, cpt color ramp data, etc), this is all just noise. And it's especially annoying on github runs where the sheer length of the build log drags the whole page to a crawl!

This change reduces the total steps from ~11800 (as above), to ~5900

(**maybe** if we're lucky it'll speed up compilations too, as eg ninja **may** be able to make more intelligent decisions when pooling jobs)

### CODE REVIEW

File: resources/CMakeLists.txt [hunk 1]
```diff
@@ -29,9 +29,10 @@ if (HAVE_OPENCL)
 endif()
 
 # Server, also install server subdirectory
+SET(SERVER_DEST_RESOURCE_FILES "")
```
Comment (m-kuhn) [inline resources/CMakeLists.txt:32]: Can we use lowercase here?
In general this is preferred in newer cmake code (I plan to do a mass reformat on uppercase ones)

Comment (m-kuhn) [reply resources/CMakeLists.txt:32]:
```suggestion
set(SERVER_DEST_RESOURCE_FILES "")
```

File: resources/CMakeLists.txt [hunk 2]
```diff
@@ -29,9 +29,10 @@ if (HAVE_OPENCL)
...
 if (WITH_SERVER)
   file(GLOB_RECURSE SERVER_RESOURCE_FILES RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} server/*)
-  set(RESOURCES_FILES ${RESOURCES_FILES} ${SERVER_RESOURCE_FILES})
+  ADD_QGIS_RESOURCES("${CMAKE_CURRENT_SOURCE_DIR}" resources SERVER_DEST_RESOURCE_FILES "${SERVER_RESOURCE_FILES}")
```
Comment (m-kuhn) [inline resources/CMakeLists.txt:35]:
```suggestion
  add_qgis_resources("${CMAKE_CURRENT_SOURCE_DIR}" resources SERVER_DEST_RESOURCE_FILES "${SERVER_RESOURCE_FILES}")
```

File: resources/CMakeLists.txt (hunk 1 above)
Comment (nyalldawson) [reply resources/CMakeLists.txt]: Sure! I was thinking of doing it to the macro too, but wanted to keep the diff small and understandable

**Review by m-kuhn — APPROVED** [review]

**Comment by nyalldawson** [comment]
Thanks @m-kuhn !

**Pull request merged**